File: backend/app/schemas/session.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel


class SessionBase(BaseModel):
    title: str
    description: str | None = None
    subject: str | None = None


class SessionCreate(SessionBase):
    pass


class SessionUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    subject: str | None = None
    is_archived: bool | None = None


class SessionResponse(SessionBase):
    id: UUID
    user_id: UUID
    is_archived: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SessionListResponse(BaseModel):
    sessions: list[SessionResponse]
    total: int

File: frontend/src/types/session.ts
export interface Session {
  id: string;
  user_id: string;
  title: string;
  description: string | null;
  subject: string | null;
  is_archived: boolean;
  created_at: string;
  updated_at: string;
}

export interface SessionCreate {
  title: string;
  description?: string;
  subject?: string;
}

export interface SessionUpdate {
  title?: string;
  description?: string;
  subject?: string;
  is_archived?: boolean;
}

export interface SessionListResponse {
  sessions: Session[];
  total: number;
}
